### PR TITLE
[BugFix] fix lake compaction threads not exit properly

### PR DESCRIPTION
**Pull request by starrocks-xupeng**
## What I'm doing:

Fixes https://github.com/StarRocks/StarRocksTest/issues/8630

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/storage/lake/tablet_manager.cpp [hunk 1]
```diff
@@ -813,4 +813,8 @@ StatusOr<SegmentPtr> TabletManager::load_segment(const FileInfo& segment_info, i
                         std::move(tablet_schema));
 }
 
+void TabletManager::close() {
```
[the remark this answers is not in the document]
Comment (starrocks-xupeng) [reply be/src/storage/lake/tablet_manager.cpp:816]: stop seems strange for TabletManager

[the remark this answers is not in the document]
Comment (kevincai) [reply be/src/storage/lake/tablet_manager.cpp:816]: keep consistent with other modules. if it doesn't apply to TabletManager, just use stop_scheduler, or tabletManager->get_compaction_scheduler->stop();

[the remark this answers is not in the document]
Comment (starrocks-xupeng) [reply be/src/storage/lake/tablet_manager.cpp]: done

File: be/src/storage/lake/compaction_scheduler.cpp [hunk 2]
```diff
@@ -157,8 +157,14 @@ CompactionScheduler::CompactionScheduler(TabletManager* tablet_mgr)
...
-    _stopped.store(true, std::memory_order_relaxed);
-    _threads->shutdown();
+    stop();
+}
+
+void CompactionScheduler::stop() {
+    if (!_stopped) {
+        _stopped.store(true);
+        _threads->shutdown();
+    }
```
Comment (kevincai) [inline be/src/storage/lake/compaction_scheduler.cpp:167]:
```suggestion
    bool expected = false;
    if (_stopped.compare_exchange_strong(&expected, true)) {
        _threads->shutdown();
    }
```

make sure the check and store ops are atomic under no lock even though it is expected no concurrent call of this `stop()` interface.

Comment (starrocks-xupeng) [reply be/src/storage/lake/compaction_scheduler.cpp]: done

**Review by kevincai — DISMISSED** [review]

**Review by kevincai — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**